File: 11-container-with-most-water/container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int maxA=0;
        int p1=0;
        int p2=height.length-1;
        while(p1<p2)
        {
            maxA=Math.max(maxA,((p2-p1)*Math.min(height[p1],height[p2])));
            if(height[p1]<height[p2])
            p1++;
            else
            p2--;
        }
        return maxA;
    }
}

File: 11-container-with-most-water/container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& h) {
        int ans=0;
        int i=0;
        int j=h.size()-1;
        while(i<j)
        {
            int x=min(h[i],h[j]);
            ans=max(ans,(int)(x*(j-i)));
            if(x==h[i])
            i++;
            else
            j--;
        }
        return ans;
    }
};